docs/DeveloperGuide.md

```diff
@@ -1438,6 +1413,340 @@ testers are expected to do more *exploratory* testing.
+
+    3. Test case (Missing index): `delete_tag_d t\Orthopaedic`<br>
+       Expected: Specialisation is not deleted. Error details shown in the status message.
+
+    4. Test case (Missing parameter): `delete_tag_d 1`<br>
+       Expected: Similar to previous.
+
+    5. Test case (Multiple parameters): `delete_tag_d 1 t\Orthopaedic t\Orthopaedic`<br>
+       Expected: Similar to previous.
+
+### Editing a doctor's remark
+
+1. Editing a doctor's remark while all doctors are being shown
+
+    1. Prerequisites: List all doctors using the `list_d` command. Multiple doctors in the doctor list.
+
+    2. Test case: `remark_d 1 r\Doctor away from 2023-11-01 to 2023-11-15`<br>
+       Expected: Remark successfully added to the first doctor. Success message shown in the status message.
+
+    3. Test case: `remark_d 1 r\` or `remark_d 1`<br>
+       Expected: Remark successfully removed from the first doctor. Success message shown in the status message.
+
+    4. Test case (Missing index): `remark_d r\Doctor away from 2023-11-01 to 2023-11-15`<br>
+       Expected: Remark is not deleted. Error details shown in the status message.
+
+### Adding an appointment
+
+1. Adding an appointment
+
+    1. Prerequisites: There exist no appointment with patient's NRIC `S8712461K`, doctor's NRIC `T1258979J` from `2023-09-11 07:30` to `2023-09-11 08:00` in the database.
+
+    2. Test case (Valid parameters): `add_a dic\T1258979J pic\S8712461K from\2023-09-11 07:30 to\2023-09-11 08:00`<br>
+       Expected: Appointment successfully added into appointment list. Details of the added appointment shown in the status message. The list of appointments now consists of the added appointment.
+
+    3. Test case (Missing parameter): `add_a dic\T1258979J pic\S8712461K`, `add_a dic\T1258979J from\2023-09-11 07:30 to\2023-09-11 08:00` or any command with missing parameters<br>
+       Expected: No appointment is added. Error details shown in the status message.
+
+    4. Test case (Invalid patient NRIC): `add_a dic\T1258979J pic\S871461K from\2023-09-11 07:30 to\2023-09-11 08:00`<br>
+       Expected: Similar to previous.
+
+    5. Test case (Invalid doctor NRIC): `add_a dic\T125899J pic\S8712461K from\2023-09-11 07:30 to\2023-09-11 08:00`<br>
+       Expected: Similar to previous.
+
+    6. Test case (Invalid start time): `add_a dic\T1258979J pic\S8712461K from\2023-13-09 07:30 to\2023-09-11 08:00`<br>
+       Expected: Similar to previous.
+
+    7. Test case (Invalid end time): `add_a dic\T1258979J pic\S8712461K from\2023-09-11 07:30 to\2023-13-11 08:00`<br>
+      Expected: Similar to previous.
+
+    8. Test case (Repeated Parameter): `add_a dic\T1258979J dic\T1258979J pic\S8712461K from\2023-09-11 07:30 to\2023-09-11 08:00` or any command with repeated parameter<br>
+       Expected: Similar to previous.
+
+### Listing appointments
+
+1. Listing all appointments
+
+    1. Prerequisites: Multiple patients in the patient list.
```

Comment:
`Multiple patients in the patient list` change to `Multiple appointments in the appointment list`